src/textual/widgets/_progress_bar.py

```diff
@@ -0,0 +1,434 @@
+    }
+    Bar > .bar--indeterminate {
+        color: $error;
+        background: $foreground 10%;
+    }
+    Bar > .bar--complete {
+        color: $success;
+        background: $foreground 10%;
+    }
+    """
+
+    _percentage: reactive[float | None] = reactive[Optional[float]](None)
+    """The percentage of progress that has been completed."""
+    _start_time: float | None
+    """The time when the widget started tracking progress."""
+
+    def __init__(
+        self,
+        name: str | None = None,
+        id: str | None = None,
+        classes: str | None = None,
+        disabled: bool = False,
+    ):
+        """Create a bar for a [`ProgressBar`][textual.widgets.ProgressBar]."""
+        super().__init__(name=name, id=id, classes=classes, disabled=disabled)
+        self._start_time = None
+        self._percentage = None
+
+    def watch__percentage(self, percentage: float | None) -> None:
+        """Manage the timer that enables the indeterminate bar animation."""
+        if percentage is not None:
+            self.auto_refresh = None
+        else:
+            self.auto_refresh = 1 / 15
+
+    def render(self) -> RenderResult:
+        """Render the bar with the correct portion filled."""
+        if self._percentage is None:
+            return self.render_indeterminate()
+        else:
+            bar_style = (
+                self.get_component_rich_style("bar--bar")
+                if self._percentage < 1
+                else self.get_component_rich_style("bar--complete")
+            )
+            return BarRenderable(
+                highlight_range=(0, self.size.width * self._percentage),
+                highlight_style=Style.from_color(bar_style.color),
+                background_style=Style.from_color(bar_style.bgcolor),
+            )
+
+    def render_indeterminate(self) -> RenderResult:
+        """Render a frame of the indeterminate progress bar animation."""
+        width = self.size.width
+        highlighted_bar_width = 0.25 * width
+        # Width used to enable the visual effect of the bar going into the corners.
+        total_imaginary_width = width + highlighted_bar_width
+
+        speed = 30  # Cells per second.
+        # Compute the position of the bar.
+        start = (speed * self._elapsed_time()) % (2 * total_imaginary_width)
+        if start > total_imaginary_width:
+            # If the bar is to the right of its width, wrap it back from right to left.
+            start = 2 * total_imaginary_width - start  # = (tiw - (start - tiw))
+        start -= highlighted_bar_width
+        end = start + highlighted_bar_width
+
+        bar_style = self.get_component_rich_style("bar--indeterminate")
+        return BarRenderable(
+            highlight_range=(max(0, start), min(end, width)),
+            highlight_style=Style.from_color(bar_style.color),
+            background_style=Style.from_color(bar_style.bgcolor),
+        )
+
+    def _elapsed_time(self) -> float:
```

Comment:
If this isn't a property. It should begin with a verb, i.e. `_get_elapsed_time`.